src/content/app/entity-viewer/gene-view/components/gene-view-sidebar/external-references/GeneExternalReferences.tsx

```diff
@@ -41,8 +41,7 @@ type ExternalReferencesGroupType = {
 };
 
 const GeneExternalReferences = () => {
-  const { entityId, genomeId } = useParams<'genomeId' | 'entityId'>();
-  const geneId = entityId ? parseFocusObjectIdFromUrl(entityId).objectId : null;
+  const { genomeId, geneId } = useGeneViewIds();
```

Comment:
can geneId be null or undefined or '' ? My understanding of the hook `useGeneViewIds` it will return undefined if there is no geneId. is that right?

Reply:
so what happen if there is no entityId from this line?
https://github.com/Ensembl/ensembl-client/blob/dev/src/content/app/entity-viewer/gene-view/hooks/useGeneViewIds.ts#L28

I am just confused with the if statement in the hook above. If entityId cannot be undefined or null then we dont need that if statement

Reply:
The way that the hook is written, it will return `undefined`.

But `GeneExternalReferences` is only mounted on the route that has an entity id in the url; so when called inside of `GeneExternalReferences`, it should return a string.